File: src/com/intellij/idea/plugin/hybris/diagram/businessProcess/impl/BpDiagramColorManagerIml.java
/*
 * This file is part of "hybris integration" plugin for Intellij IDEA.
 * Copyright (C) 2014-2016 Alexander Bartash <AlexanderBartash@gmail.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */

package com.intellij.idea.plugin.hybris.diagram.businessProcess.impl;

import com.intellij.diagram.DiagramEdge;
import com.intellij.idea.plugin.hybris.diagram.businessProcess.BpDiagramColorManager;
import com.intellij.idea.plugin.hybris.common.utils.HybrisI18NBundleUtils;
import com.intellij.ui.JBColor;
import org.apache.commons.lang3.StringUtils;

import java.awt.*;

/**
 * Created 11:11 PM 31 January 2016.
 *
 * @author Alexander Bartash <AlexanderBartash@gmail.com>
 */
public class BpDiagramColorManagerIml extends BpDiagramColorManager {

    @Override
    public Color getEdgeColor(final DiagramEdge edge) {
        final String edgeType = edge.getRelationship().toString();

        if (StringUtils.isBlank(edgeType) || "OK".equalsIgnoreCase(edgeType)) {
            return new JBColor(
                new Color(9, 128, 0),
                new Color(9, 128, 0)
            );
        }

        if ("NOK".equalsIgnoreCase(edgeType) || "ERROR".equalsIgnoreCase(edgeType) || "FAIL".equalsIgnoreCase(edgeType)) {
            return new JBColor(
                new Color(161, 49, 42),
                new Color(161, 49, 42)
            );
        }

        final String timeoutLabel = HybrisI18NBundleUtils.message("hybris.business.process.timeout");

        if (StringUtils.startsWith(edgeType, timeoutLabel)) {
            return new JBColor(
                new Color(161, 49, 42),
                new Color(161, 49, 42)
            );
        }

        return new JBColor(
            new Color(71, 71, 71),
            new Color(155, 158, 161)
        );
    }
}

File: src/com/intellij/idea/plugin/hybris/diagram/businessProcess/impl/BpDiagramColorManagerIml.kt
/*
 * This file is part of "hybris integration" plugin for Intellij IDEA.
 * Copyright (C) 2014-2016 Alexander Bartash <AlexanderBartash@gmail.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package com.intellij.idea.plugin.hybris.diagram.businessProcess.impl

import com.intellij.diagram.DiagramBuilder
import com.intellij.diagram.DiagramEdge
import com.intellij.idea.plugin.hybris.diagram.businessProcess.BpColors.CANCEL
import com.intellij.idea.plugin.hybris.diagram.businessProcess.BpColors.CYCLE
import com.intellij.idea.plugin.hybris.diagram.businessProcess.BpColors.DEFAULT
import com.intellij.idea.plugin.hybris.diagram.businessProcess.BpColors.NOK
import com.intellij.idea.plugin.hybris.diagram.businessProcess.BpColors.OK
import com.intellij.idea.plugin.hybris.diagram.businessProcess.BpColors.PARTIAL
import com.intellij.idea.plugin.hybris.diagram.businessProcess.BpColors.START
import com.intellij.idea.plugin.hybris.diagram.businessProcess.BpColors.TIMEOUT
import com.intellij.idea.plugin.hybris.diagram.businessProcess.BpDiagramColorManager

/**
 * TODO: Add user-defined project-based mapping for custom transition names
 */
class BpDiagramColorManagerIml : BpDiagramColorManager() {

    override fun getEdgeColorKey(builder: DiagramBuilder, edge: DiagramEdge<*>) = when (edge) {
        is BpDiagramFileOKEdge -> OK
        is BpDiagramFileNOKEdge -> NOK
        is BpDiagramFileStartEdge -> START
        is BpDiagramFileCancelEdge -> CANCEL
        is BpDiagramFilePartialEdge -> PARTIAL
        is BpDiagramFileCycleEdge -> CYCLE
        is BpDiagramFileTimeoutEdge -> TIMEOUT
        else -> DEFAULT
    }

}
